pkg/controller/managedresources/controller.go

```diff
@@ -151,9 +151,14 @@ func (r *Reconciler) reconcile(mr *resourcesv1alpha1.ManagedResource, log logr.L
 					continue
 				}
 
+				obj := &unstructured.Unstructured{Object: decodedObj}
+				if obj.GetKind() != "Namespace" && obj.GetNamespace() == "" {
```

Comment:
`obj.GetKind() != "Namespace"` hmm, what about for cluster level resources like `ClusterRole` ?

Reply:
Non-namespaced resources are applied but in the status are shown with ns as:
```
    - apiVersion: rbac.authorization.k8s.io/v1
      kind: ClusterRole
      name: role-grantor
      namespace: default
```

Probably @rfranzke could share what was the motivation removing the namespace defaulting in 77126f1274ae7dfb524b0c6a2a47f96514cf7e07.

Reply:
So it will be set anyway, simply the API will ignore it. @mandelsoft  @rfranzke, am I right?